src/System.Windows.Forms/src/System/Windows/Forms/BindingContext.cs

```diff
@@ -332,15 +332,15 @@ private static void CheckPropertyBindingCycles(BindingContext newBindingContext,
 
         private void ScrubWeakRefs()
         {
-            ArrayList cleanupList = null;
-            foreach (DictionaryEntry de in _listManagers)
+            List<HashKey> cleanupList = null;
+            foreach (KeyValuePair<HashKey, WeakReference> de in _listManagers)
```

Comment:
Can we do deconstruction here? E.g., something like:
```suggestion
            foreach ((HashKey key, WeakReference wRef) in _listManagers)
```